### PR TITLE
fix(ci): make CI guards fail closed (ProofGraph sig writes, release TEE mock)

### PR DESCRIPTION
**Pull request by mindburnlabs**
## Summary

Two guard steps in the kernel CI job were silently inert on `ubuntu-latest` (no `rg` installed):

- **ProofGraph signature write guard** — `rg ... || true` swallowed exit 127 along with the intended exit 1 ("no matches"), so the step passed regardless of `Node.Sig` write violations. Observed in run 30707356093, job 91388660679 (`line 10: rg: command not found`, step green).
- **Release TEE mock check** (`check_release_no_mock_tee.sh`, same job, ci.yml:170) — `if rg ...; then FAIL` treated the missing binary (exit 127) as "no matches" and printed `release TEE mock check passed`. Found during this fix; same defect class.

This matches the known estate pattern of inert enforcement paths (verifiable-execution audit).

Fix, both scripts:
- Rewrite on `git grep` — guaranteed present in any checkout, scans tracked files only, one code path for CI and local runs (no tool-presence fallback to drift).
- Explicit exit-code discipline: 0 = match (violation → exit 1), 1 = clean, >1 = scanner error → **fail closed**.
- git grep exits 1 for pathspecs that match nothing (verified on git 2.53) — indistinguishable from a clean scan — so missing scan surfaces are handled explicitly: the TEE check asserts its scan paths exist (exit 2 when missing; Codex P2 review finding), and the ProofGraph guard self-tests that the pattern still finds the allowed `Node.Sig` assignment in `core/pkg/proofgraph/node.go` before trusting an empty scan (exit 2 if the scanner machinery is inert, including that file going missing).
- `main` is clean for both patterns — verified locally, so enabling real enforcement cannot turn CI red on merge.

## Verification

Per script: clean tree → exit 0; planted violation in a tracked file → exit 1 + listing; scanner unavailable (empty PATH) → exit 2/127 fail-closed; renamed scan dir (TEE) → exit 2; shellcheck clean. Old behavior in the failure cases was exit 0 pass — that is the bug.

Linear: HELM-434.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/ci/check_release_no_mock_tee.sh [hunk 1]
```diff
@@ -4,7 +4,17 @@ set -euo pipefail
 ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
 cd "$ROOT"
 
-if rg -n 'AllowMock\s*[:=]\s*true|HELM_TEE_ALLOW_MOCK\s*[:=]\s*(1|true|yes)' .github/workflows scripts/release; then
+# Fail closed: git grep exit 0 = forbidden match, 1 = clean, >1 = scanner error
+# (including a scanned path that no longer exists). The old `if rg ...` treated
+# a missing rg binary (exit 127) as "no matches" and passed while inert.
+matches="$(git grep -nE 'AllowMock[[:space:]]*[:=][[:space:]]*true|HELM_TEE_ALLOW_MOCK[[:space:]]*[:=][[:space:]]*(1|true|yes)' -- .github/workflows scripts/release)" && rc=0 || rc=$?
```
[the remark this answers is not in the document]
Comment (mindburnlabs) [reply scripts/ci/check_release_no_mock_tee.sh]: Confirmed on git 2.53 — unmatched pathspecs exit 1 in every combination (existing+missing dir, missing-only, explicit missing file), indistinguishable from a clean scan. Fixed in 4bbffd017: scan paths are asserted to exist before grepping (exit 2 when missing), and the wrong fail-closed claim in the script comment and PR body is corrected. The ProofGraph guard is covered against the same class by its node.go self-test (missing file → exit 1 → self-test branch → exit 2). Verified: mv scripts/release aside → exit 2.

File: scripts/ci/check_release_no_mock_tee.sh [hunk 2]
```diff
@@ -4,7 +4,24 @@ set -euo pipefail
...
 
-if rg -n 'AllowMock\s*[:=]\s*true|HELM_TEE_ALLOW_MOCK\s*[:=]\s*(1|true|yes)' .github/workflows scripts/release; then
+# Fail closed: a missing scan path or a scanner error must fail the step, never
+# pass it. git grep exits 1 both for "no matches" and for pathspecs matching
+# nothing (verified on git 2.53), so path existence is asserted explicitly;
+# exit codes: 0 = forbidden match, 1 = clean, >1 = scanner error. The old
+# `if rg ...` treated a missing rg binary (exit 127) as "no matches".
+scan_paths=(.github/workflows scripts/release)
+for p in "${scan_paths[@]}"; do
+  [[ -e "${p}" ]] || { echo "scan path missing: ${p}" >&2; exit 2; }
```
[the remark this answers is not in the document]
Comment (mindburnlabs) [reply scripts/ci/check_release_no_mock_tee.sh]: Confirmed and fixed in 75f4b72ed (verified against committed states in a scratch clone, git 2.53): the scan surface must now be a real non-symlink directory (`-L`/`-d`), contain at least one tracked file (an emptied dir scans as clean), and contain no tracked symlinks (`git ls-files -s` mode 120000 — an inner symlink evades the scan the same way the dir-level one does). Each violation exits 2. Sim results: dir removal → 2, dir→symlink swap → 2, trackless dir → 2, tracked inner symlink → 2, clean → 0.

File: scripts/ci/check_release_no_mock_tee.sh [hunk 3]
```diff
@@ -4,7 +4,41 @@ set -euo pipefail
...
+    echo "scan path missing or a symlink: ${p}" >&2
+    exit 2
+  fi
+  if [[ -z "$(git ls-files -- "${p}")" ]]; then
+    echo "scan path has no tracked files: ${p}" >&2
+    exit 2
+  fi
+  # Tracked symlinks under a scan surface evade the scan the same way: git grep
+  # reads the link blob while execution follows the link target.
+  if [[ -n "$(git ls-files -s -- "${p}" | awk '$1 == "120000"')" ]]; then
```
[the remark this answers is not in the document]
Comment (mindburnlabs) [reply scripts/ci/check_release_no_mock_tee.sh]: Confirmed and fixed in dac43fac1. Rather than adding 160000 to a blacklist, the check now whitelists regular blobs (100644/100755) for every tracked entry under a scan surface — that closes the index entry-mode taxonomy outright (symlinks, gitlinks, and anything future all fail with the offending entries listed). Also added an engine self-test (a known literal must be found before an empty scan is trusted), mirroring the ProofGraph guard design. Verified against committed states: inner symlink → exit 2 (120000 listed), submodule as scan content → exit 2 (160000 listed), clean → 0.
